### PR TITLE
Build out REST Endpoints for Actions and Projects (CRUD)

### PR DESCRIPTION
**Pull request by ogwurujohnson**
#### What does this PR do?
Have CRUD on actions and projects resources endpoints for the sprint challenges REST API
API is deployed on Heroku and lives here https://project-sprint.herokuapp.com/api/
#### Description of Task to be completed?
Have the following endpoints working
`GET  /api/v1/projects`

`GET  /api/v1/projects/:id`

`POST  /api/v1/projects`

`POST  /api/v1/projects/:id/actions`

`GET  /api/v1/projects/:id/actions`

`PUT  /api/v1/projects/:id`

`DELETE  /api/v1/projects/:id`

`GET   /api/v1/actions`

`GET   /api/v1/actions/:id`

`POST   /api/v1/actions`

`PUT  /api/v1/actions/:id`

`DELETE   /api/v1/actions/:id`
#### How should this be manually tested?
After cloning the Repo, cd into it and RUN `npm install` || `yarn install`
 - Using postman test every endpoint above with this header
*Key*: `Content-Type:` *Value*: `application/json`

### CODE REVIEW

File: components/actions/actionController.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
+const Actions = require("./actionModel");
+
+const getActions = async (req, res) => {
+  try {
+    const { id } = req.params || "";
```
Comment (delbaoliveira) [inline components/actions/actionController.js]: This is good, but what happens to the actions if the { id } is an empty string?

File: middleware/index.js [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+const Actions = require("../components/actions/actionModel");
+
+async function validateProjectId(req, res, next) {
+  const { id } = req.params;
+  // checkout parseInt vs Number
+  const integerNum = Number(id);
+  if (
+    !integerNum ||
+    typeof integerNum !== "number" ||
+    Math.sign(integerNum === -1)
```
Comment (delbaoliveira) [inline middleware/index.js]: Checking if negative numbers are passed as the id. Nice 👍

File: components/actions/actionController.js (hunk 1 above)
Comment (ogwurujohnson) [reply components/actions/actionController.js]: if the id is an empty string, the middleware takes control and flags the query parameter and returns and error message

File: middleware/index.js (hunk 2 above)
Comment (ogwurujohnson) [reply middleware/index.js]: Thanks

**Pull request merged**